### PR TITLE
ddtrace/tracer: Add normalization of span duration

### PR DESCRIPTION
**Pull request by piochelepiotr**
Right now, duration can be negative, and that can cause the negative sketch to be non empty, which is not a nice behavior.

### CODE REVIEW

File: ddtrace/tracer/span.go [hunk 1]
```diff
@@ -369,9 +369,16 @@ func (s *span) finish(finishTime int64) {
 	s.context.finish()
 }
 
+func normalizeSpan(s *span) {
```
Comment (knusbaum) [inline ddtrace/tracer/span.go:372]: Do we need a separate function? Let's just inline this.

File: ddtrace/tracer/span_test.go [hunk 2]
```diff
@@ -619,6 +619,20 @@ func TestSpanLog(t *testing.T) {
 	})
 }
 
+func TestNormalizeSpan(t *testing.T) {
+	t.Run("negativeduration", func(t *testing.T) {
+		actual := span{
+			Service:  "testservice",
+			Duration: -1,
+		}
+		normalizeSpan(&actual)
```
Comment (knusbaum) [inline ddtrace/tracer/span_test.go:628]: normalizeSpan is too trivial for this to be valuable. Don't we have tests for `aggregableSpan`?

**Review by knusbaum — CHANGES_REQUESTED** [review]

File: ddtrace/tracer/span.go [hunk 3]
```diff
@@ -369,9 +369,16 @@ func (s *span) finish(finishTime int64) {
...
+	if s.Duration < 0 {
+		s.Duration = 0
+	}
+}
+
 // newAggregableSpan creates a new summary for the span s, within an application
 // version version.
 func newAggregableSpan(s *span, cfg *config) *aggregableSpan {
+	normalizeSpan(s)
```
Comment (gbbr) [inline ddtrace/tracer/span.go:381]: This doesn't seem like the right place for span normalization.

Comment (piochelepiotr) [reply ddtrace/tracer/span.go:381]: In that case, we apply normalization on both traces and stats. If that's OK, I can move it. What do you think?

Comment (gbbr) [reply ddtrace/tracer/span.go:381]: I don't see why not. A negative duration is invalid in both... Plus, they shouldn't have different field values if they represent the same exact thing...

Comment (piochelepiotr) [reply ddtrace/tracer/span.go]: sounds good, just moved it.

File: ddtrace/tracer/span.go (hunk 1 above)
Comment (gbbr) [reply ddtrace/tracer/span.go:375]: We don't need this function anymore :)

File: profiler/profile_test.go [hunk 4]
```diff
@@ -19,6 +19,7 @@ import (
 	pprofile "github.com/google/pprof/profile"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
```
Comment (gbbr) [inline profiler/profile_test.go:22]: This is the wrong order. Local imports come first, and then outside imports.

**Review by gbbr — APPROVED** [review]
LGTM. Please wait with the merge. We are in ❄️. About to release the next version.

**Review by gbbr — APPROVED** [review]
https://github.com/DataDog/dd-trace-go/pull/973#pullrequestreview-733916169.

I'll merge as soon as we release.

**Review by knusbaum — APPROVED** [review]
LGTM

**Comment by gbbr** [comment]
@piochelepiotr one small conflict to sort out before merging...

**Pull request merged**